Handle unrecognised account errors

delki8 opened the pull request:
… wallet. #347 

<!---
Hints for a successful PR:
1. It is recommended that before you submit a PR to IOTA, to open an issue first and assign yourself.
This way you may get inputs and discover parallel PRs to the one you want to submit.
2. In case of a big PR, consider breaking it up into smaller PRs. This will help to get it merged in an incremental process.
3. Note that a PR should have a *single* area of responsibility. If your PR does more than one thing than it should be split into several PRs!!!!!
4. It will be helpful if you make additional comments on the code via GitHub PR review to explain the choices you made
-->

# Description

- I found a problem that might be this one, the account name wasn't being recognized for my seed. (# How Has This Been Tested?)

Fixes #347 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

# How Has This Been Tested?

- That happened because in my daily use trinity client I gave one name to my account, let's say 'Account #1', and in the development client, for the same seed, I gave the name 'Account #2'. Then my password was correct but the seed wasn't being recoverd because the account name associated with the dev client was incorrect, the app was failing silently and what I'm doing here is to show an error to the user.


# Checklist:

- [ x ] My code follows the style guidelines for this project
- [ x ] I have performed a self-review of my own code
- [ x ] New and existing unit tests pass locally with my changes

Commit:
Desktop wallet locks and refuses to unlock. No clear way to reset the wallet. #347

- I found a problem that might be this one, the account name wasn't being recognized for my seed. That happened because in my daily use trinity client I gave one name to my account, let's say 'Account #1', and in the development client, for the same seed, I gave the name 'Account #2'. Then my password was correct but the seed wasn't being recoverd because the account name associated with the dev client was incorrect, the app was failing silently and what I'm doing here is to show an error to the user.

rajivshah3 commented on src/shared/locales/pt-BR/translation.json:
```diff
@@ -307,7 +309,7 @@
         "writeYourSeedDown": "Anotar Seed",
         "seedClearedExplanation": "Para sua segurança, a Seed foi removida da área de transferência.",
-        "letsWriteDownYourSeed": "Vamos gravar sua seed",
+        "letsWriteDownYourSeed": "Vamos salvar sua seed",
```
Please don't make manual changes to the translation files (except `locales/en/translation.json`). These are automatically generated and will likely be overwritten when translations are synced from [Crowdin](https://crowdin.com/project/iota-trinity-wallet). Instead, you can suggest changes there and they will automatically be synced back to this repository.

rajivshah3 commented on src/shared/locales/en/translation.json:
```diff
@@ -63,6 +63,8 @@
         "unrecognisedPassword": "Unrecognised password",
         "unrecognisedPasswordExplanation": "The password was not recognised. Please try again.",
+        "unrecognizedAccount": "Unrecognized account",
+        "unrecognizedAccountExplanation": "Account named {{accountName}} was not recognized. Consider reset your wallet.",
```
IOTA aims to use UK English as much as possible, so `unrecognized` should be `unrecognised` and `recognized` should be `recognised`.

delki8 commented:
It seems that I mess up... I was trying to address the #347 issue that is already opened... Sorry for this.

rajivshah3 reviewed (changes requested):
Hi, thanks for the PR! Sorry that the documentation wasn't clear on what needs to be done for localization. Please see the comments below

rajivshah3 commented:
> It seems that I mess up... I was trying to address the #347 issue that is already opened... Sorry for this.

@delki8 this seems fine except for the localization parts

Commit:
Desktop wallet locks and refuses to unlock. No clear way to reset the wallet. #347

- following project guidelines to keep using uk english

delki8 committed:
Merge branch 'develop' into bugfix/seed-associated-wrong-account-name

delki8 commented:
@rajivshah3 Ok, So I changed the name of the key to match the UK English (I didn't know that there were a difference) and changed the en/translation.json also with the UK spelling. I'm also reverting the change made directly at ptBr translation file and didn't bother to fix the other files to the UK spelling because they will be replaced either way.

And let me know if there is anything left :)

rajivshah3 commented on src/shared/locales/en/translation.json:
```diff
@@ -63,6 +63,8 @@
         "unrecognisedPassword": "Unrecognised password",
         "unrecognisedPasswordExplanation": "The password was not recognised. Please try again.",
+        "unrecognisedAccount": "Unrecognised account",
+        "unrecognisedAccountExplanation": "Account named {{accountName}} was not recognised. Consider reset your wallet.",
```
Sorry, I forgot one thing in the earlier review. Could you change the last part to `Consider resetting your wallet.` or `Please reset the wallet.`?

delki8 replied:
Not a problem.

rajivshah3 replied:
Just realized that the Advanced Settings can't be accessed if you can't log in 😅 `Please reset the wallet` might be better here in that case

delki8 replied:
hehe, I should have thought of that as well

Commit:
Desktop wallet locks and refuses to unlock. No clear way to reset the wallet. #347

- Giving a better explanation on what to do in the case of an error.

Commit:
Merge remote-tracking branch 'origin/bugfix/seed-associated-wrong-account-name' into bugfix/seed-associated-wrong-account-name

Commit:
Desktop wallet locks and refuses to unlock. No clear way to reset the wallet. #347

- Giving a better explanation on what to do in the case of an error.

rajivshah3 commented on src/shared/locales/en/translation.json:
```diff
@@ -63,6 +63,8 @@
         "unrecognisedPassword": "Unrecognised password",
         "unrecognisedPasswordExplanation": "The password was not recognised. Please try again.",
+        "unrecognisedAccount": "Unrecognised account",
+        "unrecognisedAccountExplanation": "Account named {{accountName}} was not recognised. Please go to Advanced Settings reset the wallet.",
```
I just realized Advanced Settings isn't accessible if you can't log in. Could you change the last part to `Please reset the wallet.`?

delki8 replied:
Sure, I'll do it

rajivshah3 reviewed (changes requested):
Once the string is fixed and conflicts are resolved, this should be ready to merge (pending review from @rihardsgravis)

Commit:
Merge branch 'develop' into bugfix/seed-associated-wrong-account-name

# Conflicts:
#	src/desktop/src/ui/views/onboarding/Login.js
#	src/shared/locales/cs/translation.json
#	src/shared/locales/et/translation.json
#	src/shared/locales/fr/translation.json
#	src/shared/locales/sk/translation.json
#	src/shared/locales/zh-CN/translation.json

rajivshah3 committed:
Merge branch 'bugfix/seed-associated-wrong-account-name' of https://github.com/delki8/trinity-wallet into delki8-bugfix/seed-associated-wrong-account-name

rajivshah3 committed:
Revert changes to other languages for easier PR review

rajivshah3 commented:
Thanks @delki8 ! I've asked Rihards if he can review

Commit:
Merge branch develop into bugfix/seed-associated-wrong-account-name

laumair commented:
@delki8 LGTM, thanks!   👍
